compiler/docgen.nim

```diff
@@ -556,7 +556,14 @@ $3
   if rdoccmd notin d.exampleGroups:
     d.exampleGroups[rdoccmd] = ExampleGroup(rdoccmd: rdoccmd, docCmd: d.conf.docCmd, index: d.exampleGroups.len)
   d.exampleGroups[rdoccmd].code.add "import $1\n" % outp.string.quoted
-  result = (rdoccmd, code)
+
+  var codeShown: string
+  if topLevel: # refs https://github.com/nim-lang/RFCs/issues/352
+    let title = canonicalImport(d.conf, AbsoluteFile d.filename)
+    codeShown = "import $#\n$#" % [title, code]
```

Comment:
```suggestion
    codeShown = "import $#\n\n$#" % [title, code]
```
I think it would be nicer to have a blank line between the top-level import and the rest of the code.

Reply:
this would look weird in presence of other imports, eg:
```nim
import std/tables

from std/sequtils import zip

let
  names = ["John", "Paul", "George", "Ringo"]
  years = [1940, 1942, 1943, 1940]
```

plus, vertical real estate matters to avoid too much scrolling.
if there's consensus, i can change it though, but i prefer the way i wrote it